lm_eval/models/huggingface.py

```diff
@@ -726,15 +726,11 @@ def tok_batch_encode(
 
         return encoding["input_ids"], encoding["attention_mask"]
 
-    def tok_decode(self, tokens, skip_special_tokens=True):
+    def tok_decode(self, tokens):
         if self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM:
-            return self.tokenizer.decode(
-                tokens, skip_special_tokens=skip_special_tokens
-            )
+            return self.tokenizer.decode(tokens)
         elif self.AUTO_MODEL_CLASS == transformers.AutoModelForSeq2SeqLM:
-            return self.tokenizer.decode(
-                tokens, skip_special_tokens=skip_special_tokens
-            )
+            return self.tokenizer.decode(tokens, skip_special_tokens=True)
```

Comment:
`skip_special_tokens` flag is needed for including adding the eos token to the `until` list. Otherwise it will be an empty string.